### PR TITLE
add data source for retrieving the root Organizational Unit

### PR DESCRIPTION
**Pull request by afeld**
Part of #571.

Corresponds to [this API](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListRoots.html). The motivation here is that I want to be able to create Organizational Units (OUs) from Terraform (see #4207), but to do so I need the ID of the root OU.

Acceptance tests pass:

```
$ make testacc TEST=./aws TESTARGS="-run OrganizationRoot"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run OrganizationRoot -timeout 120m
=== RUN   TestAccDataSourceAwsOrganizationRoot_basic
--- PASS: TestAccDataSourceAwsOrganizationRoot_basic (10.44s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	31.309s
```

Also tried with with an actual test case:

```hcl
data "aws_organizations_root" "root" {}

output "arn" {
  value = "${data.aws_organizations_root.root.arn}"
}

output "id" {
  value = "${data.aws_organizations_root.root.id}"
}
```

```
$ $GOPATH/bin/terraform apply
data.aws_organizations_root.root: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

arn = arn:aws:organizations::422624340815:root/o-pwk4n5w06h/r-bgtv
id = r-bgtv
```

Thanks!

### CODE REVIEW

File: aws/data_source_aws_organizations_root_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataSourceAwsOrganizationRoot_basic(t *testing.T) {
```
Comment (afeld) [inline aws/data_source_aws_organizations_root_test.go]: Do I need to worry about Organizations possibly not being enabled?

**Comment by afeld** [comment]
Decided that having a single data source for organizational units (in the future) made more sense than having a special one for roots, so changed the naming around.

File: aws/provider.go [hunk 2]
```diff
@@ -221,6 +221,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_region":                           dataSourceAwsRegion(),
 			"aws_route_table":                      dataSourceAwsRouteTable(),
 			"aws_route53_zone":                     dataSourceAwsRoute53Zone(),
+			"aws_organizations_unit":               dataSourceAwsOrganizationUnit(),
```
[the remark this answers is not in the document]
Comment (afeld) [reply aws/provider.go]: D'oh, fixed!

**Comment by afeld** [comment]
Anything else needed here?

**Comment by ndobbs** [comment]
Is this data source ready to be reviewed?

**Comment by bflad** [comment]
FYI: The commits from this pull request were included with #4207 to add the `aws_organizations_organizational_unit` resource, but later removed in that pull request which was just merged, which is why this is showing as merged. There is currently no `aws_organizations_organizational_unit` _data source_ in the Terraform AWS Provider as of right now. Please track the originating issue #571 for further information, thanks!

**Comment by breser** [comment]
@ndobbs Not sure why you're asking me and I'd missed that you had till now.  But yes I think that data source could be merged.  I was using at a previous job with a custom built copy of the aws provider.  But I'm not longer with the company that we were using it.

**Pull request merged**